run_cppcheck.sh

```diff
@@ -5,7 +5,17 @@ RESULTS_FILE=cppcheck-results.log
 # --force: force checks all define combinations (default max is 12)
 # -iaws-sdk-cpp: avoid checking AWS C++ SDK source files in our repo
 # -UWIN32: do not check WIN32-defined codepaths; this would throw errors on Mac
-cppcheck --force -iaws-sdk-cpp -UWIN32 ./src 2> ${RESULTS_FILE}
+# --check-level: use exhaustive checking for pipeline jobs; can disable during active development for quicker results
+# --inline-suppr: enable inline error suppression
+# -i: ignore CMake and installed files
+
+cppcheck \
+  --force -iaws-sdk-cpp -UWIN32 \
+  --check-level=exhaustive \
```

Comment:
Cppcheck [manual](https://cppcheck.sourceforge.io/manual.pdf) recommends enabling for "scenarios where you can wait for results" like nightly builds, and not enabling for "active development".